.github/workflows/rust.yml

```diff
@@ -35,14 +42,48 @@ jobs:
         rustflags:
 
     - name: Install Nextest test runner
-      uses: taiki-e/install-action@nextest
+      run: cargo install cargo-nextest
 
     - name: Create example env file
       run: |
         chmod +x .github/scripts/test_config.sh
         sh .github/scripts/test_config.sh
 
-    - name: Build and Run Test
+    - name: Check Formatting
+      run: cargo fmt --all -- --check
+
+    - name: Build Project
+      run: cargo build --workspace --all-features
+
+    - name: Run Tests
+      run: cargo nextest run --workspace --all-features
+
+  # Continuous Deployment Job
+  cd:
+    name: Continuous Deployment
```

Comment:
Again this name means nothing. And as Ingrid suggested, move the CD processes to a different file.